src/menu-item/menu-item.js

```diff
@@ -1,80 +1,95 @@
-import React from 'react'
-import MenuList from '../menu-list'
+import React from 'react';
+import MenuList from '../menu-list';
 
 class MenuItem extends React.Component {
-  constructor (...args) {
-    super(...args)
+  constructor(...args) {
+    super(...args);
     this.state = {
-      childHovered: false
+      childHovered: false,
+    };
+    this.handleMouseLeave = this.handleMouseLeave.bind(this);
+    this.handleMouseEnter = this.handleMouseEnter.bind(this);
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(e) {
+    e.stopPropagation();
+
+    const { name, clickItemCallback, items } = this.props;
+    if (!items) {
+      clickItemCallback(name);
     }
-    this.handleMouseLeave = (e) => this._handleMouseLeave(e)
-    this.handleMouseEnter = (e) => this._handleMouseEnter(e)
-    this.handleClick = (e) => this._handleClick(e)
   }
+  handleMouseEnter(e) {
+    this.props.mouseOverHandler(e.currentTarget);
+  }
+  handleMouseLeave(e) {
+    const currentTarget = e.currentTarget;
+    const { mouseOutHandler } = this.props;
+    mouseOutHandler(currentTarget);
+  }
+  renderInnerList() {
+    let result;
```

Comment:
We can. But default eslint settings do not allow it and I personally find it more readable.